### ISSUE TITLE
FSOD vs GFSOD

### ISSUE THREAD

**Issue by vineetrshenoy** [WuShuang1998/MFDC#5]
Hi, thank you for this work. In the repository, you have provided the configuration files for the `gfsod` evaluation scenario. If I'm not mistaken, the results in table 1 of your paper provides results for the `fsod` scenario, correct?  I have tried to reproduce your results running the code exactly as in the README, but I can not get the results in table 1.

Do you provide the configuration files for the `fsod` scenario?

**Comment by WuShuang1998** [comment]
Hi, thanks for your attention!
We don’t adopt the ‘fsod’ or ‘gfsod’ setting used in [DeFRCN](https://github.com/er-muyue/DeFRCN). In fact, the experiment setting of few-shot object detection in previous works is somewhat confused. In [DeFRCN](https://github.com/er-muyue/DeFRCN), ‘fsod’ only fine-tunes on novel classes. But in [TFA](https://github.com/ucbdrive/few-shot-object-detection), [FSRW](https://github.com/bingykang/Fewshot_Detection), [FSCE](https://github.com/megvii-research/FSCE) etc, ‘fsod’ fine-tunes on both base and novel classes, using the same training samples provided by [FSRW](https://github.com/bingykang/Fewshot_Detection), ‘gfsod’ trains for multiple runs to obtain averages and report the overall AP (base and novel).
We use the setting of the latter one ([TFA](https://github.com/ucbdrive/few-shot-object-detection)), fine-tuning on the overall classes and report the AP on novel classes.

As for the reproduced results, could you please provide more details such as environments or training logs?

**Comment by vineetrshenoy** [comment]
Thank you for your prompt response, and for clarifying how your protocol is difference from previous work.

For my environment, I have the following dependencies:

- `torch`: 1.6.0+cu101
- `torchvision`: 0.7.0+cu101
- `detectron2`: 0.3+cu101

I execute my code by calling `voc_train.sh mfdc 1` for split 1, seed 0 of the VOC training data. I train on 4 NVIDIA GEForce RTX 2080 Ti cards. The seed0 files are taken from [this link](https://github.com/bingykang/Fewshot_Detection/tree/master/data/vocsplit).

The output config and logs are at [this link](https://livejohnshopkins-my.sharepoint.com/:f:/g/personal/vshenoy4_jh_edu/Esg74S04cPpDhchZQowMZkUBszuIa_34M9ggQ-yVnxDPPg?e=Dj47Cj).

The results I get are below: 

![mfdc_results](https://user-images.githubusercontent.com/5201109/212108351-08e303e9-75c5-46a0-ae9e-3af70a1a5f6d.png)


Any insight into this would be appreciated!

**Comment by WuShuang1998** [comment]
> Thank you for your prompt response, and for clarifying how your protocol is difference from previous work.
> 
> For my environment, I have the following dependencies:
> 
> * `torch`: 1.6.0+cu101
> * `torchvision`: 0.7.0+cu101
> * `detectron2`: 0.3+cu101
> 
> I execute my code by calling `voc_train.sh mfdc 1` for split 1, seed 0 of the VOC training data. I train on 4 NVIDIA GEForce RTX 2080 Ti cards. The seed0 files are taken from [this link](https://github.com/bingykang/Fewshot_Detection/tree/master/data/vocsplit).
> 
> The output config and logs are at [this link](https://livejohnshopkins-my.sharepoint.com/:f:/g/personal/vshenoy4_jh_edu/Esg74S04cPpDhchZQowMZkUBszuIa_34M9ggQ-yVnxDPPg?e=Dj47Cj).
> 
> The results I get are below:
> 
> ![mfdc_results](https://user-images.githubusercontent.com/5201109/212108351-08e303e9-75c5-46a0-ae9e-3af70a1a5f6d.png)
> 
> Any insight into this would be appreciated!

**Comment by WuShuang1998** [comment]
> Thank you for your prompt response, and for clarifying how your protocol is difference from previous work.
> 
> For my environment, I have the following dependencies:
> 
> * `torch`: 1.6.0+cu101
> * `torchvision`: 0.7.0+cu101
> * `detectron2`: 0.3+cu101
> 
> I execute my code by calling `voc_train.sh mfdc 1` for split 1, seed 0 of the VOC training data. I train on 4 NVIDIA GEForce RTX 2080 Ti cards. The seed0 files are taken from [this link](https://github.com/bingykang/Fewshot_Detection/tree/master/data/vocsplit).
> 
> The output config and logs are at [this link](https://livejohnshopkins-my.sharepoint.com/:f:/g/personal/vshenoy4_jh_edu/Esg74S04cPpDhchZQowMZkUBszuIa_34M9ggQ-yVnxDPPg?e=Dj47Cj).
> 
> The results I get are below:
> 
> ![mfdc_results](https://user-images.githubusercontent.com/5201109/212108351-08e303e9-75c5-46a0-ae9e-3af70a1a5f6d.png)
> 
> Any insight into this would be appreciated!

Hi, it seems that there's nothing wrong with your environment. I guess the poor results are maybe due to the training samples for fine-tuning, and we indeed use the [vocsplit](https://drive.google.com/file/d/1BpDDqJ0p-fQAFN_pthn2gqiK5nWGJ-1a/view) provided by DeFRCN.
What's more, we have uploaded the training logs and base weights for split1 of voc by [Google Drive](https://drive.google.com/file/d/1NGuCd5umN9zqaqhwoAOw4c60irknk1dB/view?usp=sharing). Your can try to fine-tune the detector based on it and see if there are still problems.

**Comment by QSJZDDY** [comment]
Can you give me your training configration? Thx

**Comment by QSJZDDY** [comment]
> > 感谢您的及时回复，并澄清您的协议与以前的工作有何不同。
> > 对于我的环境，我有以下依赖项：
> > 
> > * `torch`：1.6.0+cu101
> > * `torchvision`：0.7.0+cu101
> > * `detectron2`：0.3+cu101
> > 
> > `voc_train.sh mfdc 1`我通过调用VOC 训练数据的 split 1、seed 0 来执行代码。我在 4 个 NVIDIA GEForce RTX 2080 Ti 卡上进行训练。Seed0 文件取自[此链接](https://github.com/bingykang/Fewshot_Detection/tree/master/data/vocsplit)。
> > 输出配置和日志位于[此链接](https://livejohnshopkins-my.sharepoint.com/:f:/g/personal/vshenoy4_jh_edu/Esg74S04cPpDhchZQowMZkUBszuIa_34M9ggQ-yVnxDPPg?e=Dj47Cj)。
> > 我得到的结果如下：
> > ![mfdc_结果](https://user-images.githubusercontent.com/5201109/212108351-08e303e9-75c5-46a0-ae9e-3af70a1a5f6d.png)
> > 任何对此的见解将不胜感激！
> 
> 你好，看来你的环境没有问题。我猜结果不佳可能是由于训练样本进行了微调，我们确实使用了DeFRCN提供的[vocsplit 。](https://drive.google.com/file/d/1BpDDqJ0p-fQAFN_pthn2gqiK5nWGJ-1a/view) 此外，我们还通过[Google Drive](https://drive.google.com/file/d/1NGuCd5umN9zqaqhwoAOw4c60irknk1dB/view?usp=sharing)上传了 voc split1 的训练日志和基本权重。您可以尝试在此基础上微调探测器。我希望这有帮助。

Can you tell me how to use the “eval-only” on your baseweight?
